File: src/main/java/com/vapitea/datacollector/model/DataSource.java
package com.vapitea.datacollector.model;


import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DataSource {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private String uuid;
  private String name;
  private String location;
  @Lob
  private String description;


  @Builder.Default
  @OneToMany(mappedBy = "dataSource", cascade = CascadeType.ALL, orphanRemoval = true)
  private List<Parameter> parameters = new ArrayList<>();

  @ManyToOne
  private Team team;

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    DataSource that = (DataSource) o;
    return id.equals(that.id);
  }

  @Override
  public int hashCode() {
    return 31;
  }
}

File: src/app/model/DataSource.ts
import {Team} from "./Team";
import {Parameter} from "./Parameter";

export class DataSource {
  id: number;
  uuid: string;
  name: string;
  location: string;
  description: string;
  parameters: Parameter[];
  team: Team;


  constructor(id: number, uuid: string, name: string, location: string, description: string, parameters: Parameter[], team: Team) {
    this.id = id;
    this.uuid = uuid;
    this.name = name;
    this.location = location;
    this.description = description;
    this.parameters = parameters;
    this.team = team;
  }
}
